qutip/solver/options.py

```diff
@@ -0,0 +1,215 @@
+    """
+    options = {
+        # Integration method (default = 'adams', for stiff 'bdf')
+        "method": 'adams',
+
+        "rhs": '',
+
+        # Absolute tolerance (default = 1e-8)
+        "atol": 1e-8,
+        # Relative tolerance (default = 1e-6)
+        "rtol": 1e-6,
+        # Maximum order used by integrator (<=12 for 'adams', <=5 for 'bdf')
+        "order": 12,
+        # Max. number of internal steps/call
+        "nsteps": 2500,
+        # Size of initial step (0 = determined by solver)
+        "first_step": 0,
+        # Max step size (0 = determined by solver)
+        "max_step": 0,
+        # Minimal step size (0 = determined by solver)
+        "min_step": 0,
+        # tidyup Hamiltonian before calculation (default = True)
+        "tidy": True,
+        "Operator_data_type": "input",
+        "State_data_type": "dense",
+        # Normalize the states received in feedback_args
+        "feedback_normalize": True,
+    }
+    extra_options = set()
+
+
+@optionsclass("results", SolverOptions)
+class SolverResultsOptions:
+    """
+    Class of options for evolution solvers such as :func:`qutip.mesolve` and
+    :func:`qutip.mcsolve`. Options can be specified either as arguments to the
+    constructor::
+
+        opts = SolverOptions(order=10, ...)
+
+    or by changing the class attributes after creation::
+
+        opts = SolverOptions()
+        opts.order = 10
+
+    Returns options class to be used as options in evolution solvers.
+
+    The default can be changed by::
+
+        qutip.settings.solver['order'] = 10
+
+    Options
+    -------
+    store_final_state : bool {False, True}
+        Whether or not to store the final state of the evolution in the
+        result class.
+    store_states : bool {False, True}
+        Whether or not to store the state vectors or density matrices in the
+        result class, even if expectation values operators are given. If no
+        expectation are provided, then states are stored by default and this
+        option has no effect.
+    """
+    #average_states : bool {False}
+    #    Average states values over trajectories in stochastic solvers.
+    #average_expect : bool {True}
+    #    Average expectation values over trajectories for stochastic solvers.
+    options = {
+        # store final state?
+        "store_final_state": False,
+        # store states even if expectation operators are given?
+        "store_states": False,
+        # Normalize output of solvers
+        # (turned off for batch unitary propagator mode)
+        "normalize_output": "ket",
+
+        # Average expectation values over trajectories (default = True)
+        # "average_expect": True,
+        # average expectation values
+        # "average_states": False,
+        # average mcsolver density matricies assuming steady state evolution
+        # "steady_state_average": False,
+    }
+
+
+@optionsclass("mcsolve", SolverOptions)
+class McOptions:
```

Comment:
Perhaps this should be called `McsolveOptions` to make it more clear what the class if for. I initially thought it was a generic options class of some sort.

Reply:
Seems like a good idea